build-from-security.py

```diff
@@ -42,12 +42,33 @@ def fetch_source_from_security(release, pkgname):
         # fetch/extract source (python-apt uses dpkg-source -x interally)
         pkg_src_dir = tempfile.mkdtemp(prefix="{}-{}-src".format(release, pkgname))
         os.makedirs(pkg_src_dir, exist_ok=True)
-        pkg = cache["fontconfig"]
+        pkg = cache[pkgname]
         if not glob.glob("{}/{}-*".format(pkg_src_dir, pkgname)):
             pkg.candidate.fetch_source(destdir=pkg_src_dir)
     return pkg_src_dir
 
 
+def cleanup_self_build_fontconfig():
+    with tempfile.TemporaryDirectory(prefix="freetype-cleanup") as tmp:
+        subprocess.check_call("sudo", "apt-get", "remove", "-y", "libfreetype6-dev", cwd=tmp)
+        subprocess.check_call("sudo", "apt-get", "download", "libfreetype6", cwd=tmp)
+        subprocess.check_call("sudo", "dpkg", "-i", "./libfreetype6_*.deb", cwd=tmp)
+
+
+def build_freetype(release):
+    pkgbasesrcdir = fetch_source_from_security(release, "libfreetype6-dev")
+    atexit.register(shutil.rmtree, pkgbasesrcdir)
+    pkgsrcdir = glob.glob(pkgbasesrcdir+"/freetype-*")[0]
+    # XXX: apply any patches
```

Comment:
mmm I think we should really try to apply the patches, it seems as if there are some material patches which fix real bugs, but then again I'm not a freetype expert, but the patches seem large...

Reply:
I would expect dpkg-buildpackage to apply any patches included in the Debian source package.  I think this comments refers to hypothetical additional patches on top of those?

Reply:
ah fair enough, nvm me then

Reply:
Yes, thanks @mgedmin for answer this. It still sounds like I should clarify my comment in the code a bit more :)